drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/commands/actualcommands/RightPanelMenuCommand.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.drools.workbench.screens.scenariosimulation.client.commands.actualcommands;
+
+import javax.enterprise.context.Dependent;
+
+import org.drools.workbench.screens.scenariosimulation.client.commands.ScenarioSimulationContext;
+import org.uberfire.client.mvp.PlaceStatus;
+
+@Dependent
+public class RightPanelMenuCommand extends AbstractScenarioSimulationCommand {
+
+    @Override
+    protected void internalExecute(ScenarioSimulationContext context) {
+        if (PlaceStatus.OPEN.equals(context.getPlaceManager().getStatus(context.getRightPanelRequest()))) {
```

Comment:
Is this equivalent to `context.isOpenDock`?

Reply:
@danielezonca 
No, it is not. Some of the context' properties are the one required to execute the command, others are the widgets/presenters/etc.
Inside 3347 I've moved the former to a specific inner class.